registry/store/external_registry.go

```diff
@@ -27,15 +27,24 @@ type ExternalRegistry interface {
 	ListServiceInstancesByName(namespace auth.Namespace, name string) (map[string]*ServiceInstance, error)
 	ListAllServiceInstances(namespace auth.Namespace) (map[string]ServiceInstanceMap, error)
 	InsertServiceInstance(namespace auth.Namespace, instance *ServiceInstance) error
-	DeleteServiceInstance(key string) (int, error)
+	DeleteServiceInstance(namespace auth.Namespace, instanceID string) (int, error)
 }
 
+const (
+	registryRecordKey = "reg"
+	namespaceKey      = "ns"
+	instanceKey       = "inst"
+	keySeparator      = ":"
+)
+
 // DBKey represents the service instance key
 type DBKey struct {
 	Namespace  string
 	InstanceID string
 }
 
+// Database key format:
+// reg:inst:<instance id>:ns:<namespace>
 func (dbk *DBKey) String() string {
-	return fmt.Sprintf("%s%s%s", dbk.InstanceID, keySeparator, dbk.Namespace)
+	return fmt.Sprintf("%s%s%s%s%s%s%s%s%s", registryRecordKey, keySeparator, instanceKey, keySeparator, dbk.InstanceID, keySeparator, namespaceKey, keySeparator, dbk.Namespace)
```

Comment:
So I changed the DB key to be a JSON string which makes parsing the key read from the db way easier (just an Unmarshal vs. splitting the string based on the separators - the concern being handling separators that may appear in the values).  This should make adding/removing/moving fields in the key trivial compared to maintaining a parsing function.  Let me know what you think of the change.  If you prefer it the other way, I can change it back.